test/e2e/util.go

```diff
@@ -214,20 +234,114 @@ func rootCredentials(kubeClient client.Client, name string) (map[string][]byte,
 	return secret.Data, nil
 }
 
-func lookupCNAMEMiekg(host, server string) ([]string, error) {
-	c := dns.Client{}
-	m := dns.Msg{}
-	m.SetQuestion(host+".", dns.TypeCNAME)
+func customResolver(nameserver string) *net.Resolver {
+	return &net.Resolver{
+		PreferGo: true,
+		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+			d := net.Dialer{
+				Timeout: dialTimeout,
+			}
+			if len(nameserver) > 0 {
+				return d.DialContext(ctx, network, fmt.Sprintf("%s:53", nameserver))
+			}
+			return d.DialContext(ctx, network, address)
+		},
+	}
+}
+
+func assertIngressControllerDeleted(t *testing.T, cl client.Client, ing *operatorv1.IngressController) {
+	t.Helper()
+	if err := deleteIngressController(t, cl, ing, 2*time.Minute); err != nil {
+		t.Fatalf("WARNING: cloud resources may have been leaked! failed to delete ingresscontroller %s: %v", ing.Name, err)
```

Comment:
Which cluster resources you refer to?

Reply:
means which ingress Controller are we deleting?
so here we are passing the ing from the caller, so this will be removed.

Reply:
The ingress controller resource cleanup is up to the `cluster-ingress-controller`, if there is any cleanup to do a finalize is used so that when the delete API call is done we are sure the controller is taken care by the operator. Also, the ingress controller you create is a `HostNetwork` one, so it doesn't even allocate a `LoadBalancer`.
So, I don't think we need `assure` method here, we can remove the ingress controller instance like any other by just calling `defer <delete-ingress-controller-api-call>` like for other resources (externalDNS, route, etc.).

Reply:
Done.